api/filter/network/tcp_proxy.proto

```diff
@@ -109,4 +109,10 @@ message TcpProxy {
   // The maximum number of unsuccessful connection attempts that will be made before
   // giving up. If the parameter is not specified, 1 connection attempt will be made.
   google.protobuf.UInt32Value max_connect_attempts = 7 [(validate.rules).uint32.gte = 1];
+
+  // The idle timeout for connections managed by the TCP proxy filter. The idle timeout
+  // is defined as the period in which there are no bytes sent or received on either
+  // the upstream or downstream connection. If not set, connections will never be closed
+  // by the TCP proxy due to being idle.
+  google.protobuf.Duration idle_timeout = 8 [(validate.rules).duration.gt = {}];
```

Comment:
Can you move this up next to the other timeouts before merging? I think it's less bad to have out-of-order field numbering than documentation that isn't logically grouped.

Reply:
Yeah, I wasn't sure which evil was preferred. I'll move it.